datafusion/optimizer/src/type_coercion.rs

```diff
@@ -103,7 +103,13 @@ fn optimize_internal(
             // ensure aggregate names don't change:
             // https://github.com/apache/arrow-datafusion/issues/3555
             if matches!(expr, Expr::AggregateFunction { .. }) {
-                if let Some((alias, name)) = original_name.zip(expr.name().ok()) {
+                if let Some(alias) = original_name {
+                    let name = expr
+                        .canonical_name()
+                        // TODO remove this hack - there is a difference between `expr.name()`
```

Comment:
yeah, there is something wrong with the projection pushdown logic inadvertently deciding the aggregate expression's output is not needed. I am trying to find out why